app/models/post.rb

```diff
@@ -1,5 +1,15 @@
 class Post < ApplicationRecord
   belongs_to :user
-  has_many :likes
-  has_many :comments
+  has_many :likes, dependent: :destroy
+  has_many :comments, dependent: :destroy
+
+  after_create :update_counter
```

Comment:
## Optional
- [ ] I suggest that you use [Rails active callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) `after_save` to call the `update_counter` method every time a new post has been saved to the database, then make the `update_counter` method private